reflex/state.py

```diff
@@ -1292,20 +1531,40 @@ def _mark_dirty(self):
         # have to mark computed vars dirty to allow access to newly computed
         # values within the same ComputedVar function
         self._mark_dirty_computed_vars()
+        self._mark_dirty_substates()
 
-        # Propagate dirty var / computed var status into substates
+    def _mark_dirty_substates(self):
+        """Propagate dirty var / computed var status into substates."""
         substates = self.substates
         for var in self.dirty_vars:
             for substate_name in self._substate_var_dependencies[var]:
                 self.dirty_substates.add(substate_name)
-                if substate_name not in substates:
-                    continue
                 substate = substates[substate_name]
                 substate.dirty_vars.add(var)
                 substate._mark_dirty()
 
+    def _get_was_touched(self) -> bool:
```

Comment:
Should this be called `set_was_touched`? I got confused below on line 1566 when you called it, as I didn't realize this also sets the value. The `get` is just the attribute itself right?

Reply:
Yeah i went back and forth on what to call this one... The flag is called `_was_touched`, and this function `_get_was_touched` checks both the flag and other conditions (and then updates the flag when those conditions indicate touching has occurred). The reason you would call this function is to decide whether the state has been modified since it was instantiated. I'll split the getting and updating logic out to clarify.